cvat-sdk/cvat_sdk/core/uploading.py

```diff
@@ -273,17 +277,20 @@ def upload_file_and_wait(
     ):
         url = self._client.api_map.make_endpoint_url(endpoint.path, kwsub=url_params)
         params = {"format": format_name, "filename": filename.name}
-        self.upload_file(
+        response = self.upload_file(
             url, filename, pbar=pbar, query_params=params, meta={"filename": params["filename"]}
         )
 
+        rq_id = json.loads(response.data).get("rq_id")
+        assert rq_id, "The rq_id was not found in the response"
+
         self._wait_for_completion(
             url,
             success_status=201,
             positive_statuses=[202],
             status_check_period=status_check_period,
             query_params=params,
-            method="POST",
+            method="PUT",
```

Comment:
Why? It seems wrong, why would this use `PUT` when the initial request was `POST`?

Reply:
Because the POST request means that we will create the rq_job (only such an approach will allow us to determine the presence of zombie rq_jobs). We have the same logic on the UI client with PUT request.